src/extensions/layout-selector/index.js

```diff
@@ -117,6 +140,80 @@ class LayoutSelector extends Component {
 		} );
 	}
 
+	detectImageBlocks( clientIds ) {
+		const {
+			getBlockName,
+			getBlockAttributes,
+		} = this.props;
+
+		const imageBlockTypes = [ 'core/image', 'core/cover' ];
+		const galleryBlockTypes = [ 'core/gallery' ];
+
+		return clientIds.map( ( clientId ) => {
+			const blockName = getBlockName( clientId );
+			const blockAttributes = getBlockAttributes( clientId );
+
+			if ( imageBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'id', 'url' ] ) };
+			}
+
+			if ( galleryBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'ids', 'images' ] ) };
+			}
+		} );
+	}
+
+	uploadExternalImages( clientId, blockAttributes ) {
+		const {
+			getBlockAttributes,
+			mediaUpload,
+			updateBlockAttributes,
+		} = this.props;
+
+		const urls = [];
+
+		if ( blockAttributes.hasOwnProperty( 'images' ) ) {
+			blockAttributes.images.forEach( ( image ) => urls.push( image.url ) );
+		} else {
+			urls.push( blockAttributes.url );
+		}
+
+		urls.forEach( ( imageUrl, index ) => {
+			window.fetch( imageUrl )
+				.then( ( response ) => response.blob() )
+				.then( ( blob ) => {
+					mediaUpload( {
+						filesList: [ blob ],
+						allowedTypes: [ 'image' ],
+						onFileChange( [ media ] ) {
+							if ( blockAttributes.hasOwnProperty( 'images' ) ) {
+								const currentAttributes = getBlockAttributes( clientId );
+
+								const newImages = currentAttributes.images.map( ( oldImage, oldIndex ) => {
+									return oldIndex === index
+										? Object.assign( {}, oldImage, { url: media.url } )
```

Comment:
I would use the slightly less opaque syntax here.
```js
{
 ...oldImage,
url: media.url
}
```

Reply:
Fixed in 07f1e16